server/game/cards/08.2-JtO/ArianneMartell.js

```diff
@@ -0,0 +1,23 @@
+const DrawCard = require('../../drawcard.js');
+
+class ArianneMartell extends DrawCard {
+    setupCardAbilities(ability) {
+        this.action({
+            title: 'Return character to hand',
+            cost: ability.costs.returnSelfToHand(),
+            target: {
+                cardCondition: card => card.location === 'play area' && card.getType() === 'character' &&
+                                       card.getStrength() < this.getStrength()
```

Comment:
You need to get the snapshotted strength for Arianne instead of just `this.getStrength()`, since returning her to hand will reset her strength to always be the printed value.